### PR TITLE
feat(atomic,headless): add more out of the box analytics configuration for atomic

### PR DESCRIPTION
**Pull request by olamothe**
Most notably `documentLocation`, `documentReferrer`, which Headless cannot determine itself (needs to be in a browser context).

Added auto resolving of Facet title (Headless does not contain the notion of facet label: it's purely a UI option).

Some minor changes in Headless for analytics parameters: Instead of sending an empty string when a value is un-available, simply don't send anything.

https://coveord.atlassian.net/browse/KIT-1623

### CODE REVIEW

**Review by ThibodeauJF — APPROVED** [review]

File: packages/atomic/src/components/atomic-search-interface/analytics-config.spec.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
+import {
```
Comment (liviarett) [inline packages/atomic/src/components/atomic-search-interface/analytics-config.spec.ts:1]: there are quite a few warnings in this file, could they be avoided?

File: packages/atomic/src/components/atomic-search-interface/analytics-config.ts [hunk 2]
```diff
@@ -4,40 +4,99 @@ import {
...
-  enabled: boolean
+  enabled: boolean,
+  store: ObservableMap<AtomicStore>
 ): AnalyticsConfiguration {
+  const analyticsClientMiddleware = (
+    event: string,
+    payload: AnalyticsPayload
+  ) => augmentAnalytics(event, payload, store, searchEngineConfig);
+
+  const outOfTheBoxConfig: AnalyticsConfiguration = {
```
Comment (liviarett) [inline packages/atomic/src/components/atomic-search-interface/analytics-config.ts:22]: do you guys always use 'out of the box' in names, or would it be better to change to `default`?

**Review by liviarett — APPROVED** [review]

File: packages/atomic/src/components/atomic-search-interface/analytics-config.ts (hunk 2 above)
Comment (btaillon) [reply packages/atomic/src/components/atomic-search-interface/analytics-config.ts:22]: I would also prefer `default` or `fallback`.

**Review by btaillon — APPROVED** [review]

File: packages/atomic/src/components/atomic-search-interface/analytics-config.spec.ts (hunk 1 above)
Comment (olamothe) [reply packages/atomic/src/components/atomic-search-interface/analytics-config.spec.ts:2]: Well add an eslint disable for `any`. Unfortunately, it's needed in this file. The types coming from `coveo.analytics` for the analytics middleware uses `any`.

File: packages/atomic/src/components/atomic-search-interface/analytics-config.ts (hunk 2 above)
Comment (liviarett) [reply packages/atomic/src/components/atomic-search-interface/analytics-config.ts]: yeah sorry, I meant replacing `outOfTheBox` with `default`, so `defaultConfig`

**Pull request merged**